src/Workspaces/Core/MSBuild/MSBuild/VisualBasic/VisualBasicProjectFile.cs

```diff
@@ -0,0 +1,468 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.MSBuild;
+using Microsoft.CodeAnalysis.MSBuild.Build;
+using Microsoft.CodeAnalysis.MSBuild.Logging;
+using Roslyn.Utilities;
+using MSB = Microsoft.Build;
+
+namespace Microsoft.CodeAnalysis.VisualBasic
+{
+    internal class VisualBasicProjectFile : ProjectFile
+    {
+        public VisualBasicProjectFile(VisualBasicProjectFileLoader loader, MSB.Evaluation.Project loadedProject, ProjectBuildManager buildManager, DiagnosticLog log)
+            : base(loader, loadedProject, buildManager, log)
+        {
+        }
+
+        protected override SourceCodeKind GetSourceCodeKind(string documentFileName)
+        {
+            // TODO: uncomment when fixing https://github.com/dotnet/roslyn/issues/5325
+            //return documentFileName.EndsWith(".vbx", StringComparison.OrdinalIgnoreCase)
+            //    ? SourceCodeKind.Script
+            //    : SourceCodeKind.Regular;
+            return SourceCodeKind.Regular;
+        }
+
+        public override string GetDocumentExtension(SourceCodeKind sourceCodeKind)
+        {
+            // TODO: uncomment when fixing https://github.com/dotnet/roslyn/issues/5325
+            //return (sourceCodeKind != SourceCodeKind.Script) ? ".vb" : ".vbx";
+            return ".vb";
+        }
+
+        protected override IEnumerable<MSB.Framework.ITaskItem> GetCommandLineArgsFromModel(MSB.Execution.ProjectInstance executedProject)
+        {
+            return executedProject.GetItems("VbcCommandLineArgs");
+        }
+
+        protected override ProjectFileInfo CreateProjectFileInfo(MSB.Execution.ProjectInstance project)
+        {
+            var projectDirectory = GetProjectDirectory(project);
+
+            var commandLineArgs = this.GetCommandLineArgsFromModel(project)
+                .Select(item => item.ItemSpec)
+                .Where(item => item.StartsWith("/"))
```

Comment:
>      .Where(item => item.StartsWith("/")) [](start = 10, length = 42)

Can't command line options be specified with a - as well?

Reply:
Does the MSBuild task ever set them that way?

Reply:
AFAICT, the MSBuildTask never sets command-line args with `-`. Do let me know if I'm wrong about that.